src/hpc/thread.c

```diff
@@ -1309,73 +1309,91 @@ Region * CurrentRegion(void)
     return TLS(currentRegion);
 }
 
-#ifdef VERBOSE_GUARDS
-
-static void PrintGuardError(char *       buffer,
-                            char *       mode,
-                            Obj          obj,
-                            const char * file,
-                            unsigned     line,
-                            const char * func,
-                            const char * expr)
-{
-    sprintf(buffer, "No %s access to object %llu of type %s\n"
-                    "in %s, line %u, function %s(), accessing %s",
-            mode, (unsigned long long)(UInt)obj, TNAM_OBJ(obj), file, line,
-            func, expr);
-}
-void WriteGuardError(Obj          o,
-                     const char * file,
-                     unsigned     line,
-                     const char * func,
-                     const char * expr)
-{
-    char * buffer = alloca(strlen(file) + strlen(func) + strlen(expr) + 200);
-    ImpliedReadGuard(o);
-    if (TLS(DisableGuards))
-        return;
-    SetGVar(&LastInaccessibleGVar, o);
-    PrintGuardError(buffer, "write", o, file, line, func, expr);
-    ErrorMayQuit("%s", (UInt)buffer, 0);
-}
+// Kernel debugging information for guards.
+//
+// When compiled with -DDEBUG_GUARDS, the GAP variable GUARD_ERROR_STACK
+// will contain the C stack after an error, allowing us to located where
```

Comment:
```suggestion
// is set to a GAP plain list describing the C stack after an error, allowing us to located where
```
Actually, might be better to rephrase: "When DEBUG_GUARDS is #defined and an error occurs, then the GAP variable ..."